packages/flame/lib/src/game/base_game.dart

```diff
@@ -87,13 +87,34 @@ class BaseGame extends Game with FPSCounter {
     _combinedProjector = Projector.compose([camera, viewport]);
   }
 
+  /// This method setps up the OrderedSet instance used by this game, before
+  /// any lifecycle methods happen.
+  ///
+  /// You can return a specific sub-class of OrderedSet, like
+  /// [QueryableOrderedSet] for example, that we use for Collidables.
+  OrderedSet<Component> createOrderedSet() {
+    final comparator = Comparing.on<Component>((c) => c.priority);
+    if (this is HasCollidables) {
+      final qos = QueryableOrderedSet<Component>(comparator);
+      qos.register<Collidable>();
+      return qos;
+    }
+    return OrderedSet<Component>(comparator);
+  }
+
   /// This method is called for every component added.
   /// It does preparation on a component before any update or render method is called on it.
   ///
   /// You can use this to setup your mixins, pre-calculate stuff on every component, or anything you desire.
   /// By default, this calls the first time resize for every component, so don't forget to call super.preAdd when overriding.
   @mustCallSuper
   void prepare(Component c) {
+    if (c is Collidable) {
```

Comment:
I am not sure why this check was not consistent with the others. Lmk @spydon if there was a reason.

Reply:
That's odd, feels like I must have had a reason, but they look equivalent to me.